src/field/pokemon.ts

```
@@ -3523,28 +3542,52 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return false;
   }
 
-  getTag<T extends BattlerTag = BattlerTag>(...tagTypes: BattlerTagType[]): T | undefined {
+  /**
+   * @param tagTypes - The {@linkcode BattlerTagType}s to retrieve
+   * @returns The first {@linkcode BattlerTag} found of the specified type(s), or `undefined` if none is found
+   */
+  public getTag<T extends BattlerTag = BattlerTag>(...tagTypes: BattlerTagType[]): T | undefined {
     return this.summonData.tags.find((t) => tagTypes.includes(t.tagType)) as T | undefined;
   }
 
   /**
    * Helper function to check if a Pokemon has any of the input tag types.
    * @param tagTypes - The battler tag types to search for
-   * @returns `true` if the Pokemon has at least one of the battler tags, `false` otherwise
+   * @returns Whether the Pokemon has at least one of the input battler tags
    */
   public hasTag(...tagTypes: BattlerTagType[]): boolean {
+    if (tagTypes.length === 0) {
+      console.warn("`Pokemon#hasTag` called with no parameters!");
+      console.trace();
```

Should those stay? or were they just debugging?

---

Not sure what to do about this. `.hasTag` should never be called with no params passed in because that's pointless, but removing the rest param would be disruptive. Is there a way to make TypeScript complain if nothing is passed to the function without changing the params? `Pokemon#hasTag(tagType, ...tagTypes)` doesn't work either because then you can't do `.hasTag(...SOME_TAG_TYPES)`.

---

I would just throw an Error, or return `false` with a warning (no trace?). Anything else would require much more unnecessary refactoring

---

Well I want the trace so we can find out where it happened and fix the mistake. Although I guess it'll be easy to ctrl+f `hasTag()`, so maybe it's unnecessary.

---

Having `console.trace` in production seems odd to me. 
I'd Rather have an error thrown.